### PR TITLE
Add reduce option to F.ConnectionistTemporalClassification

### PR DESCRIPTION
**Pull request by delta2323**
This PR add `reduce` option to `F.ConnectionistTemporalClassification` and `F.connectionist_temporal_classification`
Related to #2558

### CODE REVIEW

**Comment by unnonouno** [comment]
Is it not compatible with the original?

**Comment by delta2323** [comment]
Could you explain what you mean more?

File: chainer/functions/loss/ctc.py [hunk 1]
```diff
@@ -269,9 +285,16 @@ def connectionist_temporal_classification(
         label_length (Variable): Length of valid sequence for each of mini
             batch ``t`` (optional). If label_length is skipped, It regards that
             all of ``t`` is valid input.
+        recude (str): Reduction option. Its value must be either
```
Comment (unnonouno) [inline chainer/functions/loss/ctc.py]: reduce

File: tests/chainer_tests/functions_tests/loss_tests/test_ctc.py [hunk 2]
```diff
@@ -52,24 +56,26 @@ def check_forward(self, t_data, xs_data, l_length, x_length):
...
+        loss_expect = xp.zeros((batch_size,), dtype=xp.float32)
+        for i in range(batch_size):
+            xtb, lb, xlb, plb = xt[i], self.l[i], x_length[i], path_length[i]
+            loss_expect[i] = -math.log(
                 self.alpha(xtb, lb, int(xlb - 1), int(plb - 1)) +
                 self.alpha(xtb, lb, int(xlb - 1), int(plb - 2)))
-        loss_expect /= batch_size
-        self.assertAlmostEqual(loss_expect, loss_value, places=5)
+        if self.reduce == 'mean':
+            loss_expect = xp.sum(loss_expect) / batch_size
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/loss_tests/test_ctc.py]: `xp.mean`?

File: chainer/functions/loss/ctc.py [hunk 3]
```diff
@@ -226,9 +232,10 @@ def forward(self, inputs):
             log_yseq, self.input_length, t,
             label_length, self.path, self.path_length, xp)
 
-        loss = utils.force_array(xp.sum(
-            _logsumexp(self.prob_trans[0], xp, axis=1)))
-        loss /= -batch_size
+        loss = -_logsumexp(self.prob_trans[0], xp, axis=1)
+        if self.reduce == 'mean':
+            loss = utils.force_array(xp.sum(loss))
```
Comment (unnonouno) [inline chainer/functions/loss/ctc.py]: How about using `xp.mean`?

Comment (delta2323) [inline chainer/functions/loss/ctc.py]: fix

File: chainer/functions/loss/ctc.py (hunk 1 above)
Comment (delta2323) [inline chainer/functions/loss/ctc.py]: Fix

File: tests/chainer_tests/functions_tests/loss_tests/test_ctc.py (hunk 2 above)
Comment (delta2323) [inline tests/chainer_tests/functions_tests/loss_tests/test_ctc.py]: Fix

**Comment by unnonouno** [comment]
LGTM
